cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/program/ProgramStatePublisher.java

```diff
@@ -26,6 +27,7 @@
 /**
  * Publishing program state messages and heartbeats
  */
+@ImplementedBy(MessagingProgramStatePublisher.class)
```

Comment:
how come this is needed vs binding it in a guice module?

Reply:
Yes, I was considering the annotation vs module. Let me move to module to be consistent

Reply:
Done